supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchJob.scala

```diff
@@ -195,32 +187,21 @@ final case class AwsBatchJob(jobDescriptor: BackendJobDescriptor,           // W
                               .`type`(JobDefinitionType.CONTAINER)
                               .build
 
-    var arn = "" // Use of null here due to underlying submitJob call
-    for (inx <- 1 to 6) {
-      try {
-        if (arn == "") {
-          arn = client.registerJobDefinition(definitionRequest).jobDefinitionArn
-        }
-      } catch {
-        case e: ClientException if e.statusCode() == 404 && inx < 6 => {
-          // RegisterJobDefinition throws 404s every once in a while
-          Log.warn(s"RegisterJobDefinition 404 - retrying (${inx}/5)")
-          Thread.sleep(100L * (inx ^ 2L))
-        }
-        case e: ClientException if e.statusCode() == 409 => {
-          // This could be a problem here, as we might have registerJobDefinition
-          // but not describeJobDefinitions permissions. We've changed this to a
-          // warning as a result of this potential
-          Log.warn("Job definition already exists. Performing describe and retrieving latest revision")
-          val defs = client
-            .describeJobDefinitions(DescribeJobDefinitionsRequest.builder().jobDefinitionName(sanitize(name)).build())
-            .jobDefinitions()
-          // We need latest revision
-          defs.get(defs.size() - 1).jobDefinitionArn()
-        }
-      }
+    val submit = async.delay(client.registerJobDefinition(definitionRequest).jobDefinitionArn)
+
+    val retry: F[String] = Scheduler.fromScheduledExecutorService(ses).retry(submit, 0.millis, _ * 2, 6, {
```

Comment:
ToL: Even if it's a little more code I personally prefer pulling out those "magic" numbers and functions into declared variables and non-anonymous function for clarity